libuuid from util-linux hides the w32api libuuid.a, which serves a 
completely different purpose.

--- origsrc/ruby-1.8.7-p352/ext/win32ole/extconf.rb	2010-11-22 01:22:26.000000000 -0600
+++ src/ruby-1.8.7-p352/ext/win32ole/extconf.rb	2011-10-26 19:21:29.696817700 -0500
@@ -35,5 +35,7 @@ end
 case RUBY_PLATFORM
 when /mswin32/
   $CFLAGS += ' /W3'
+when /cygwin/
+  $LDFLAGS += ' -L/usr/lib/w32api'
 end
 create_win32ole_makefile
